packages/visualizations-react/stories/Table/PaginatedTemplates.tsx

```diff
@@ -0,0 +1,132 @@
+    const endIndex = startIndex + size;
+    await setTimeout(() => {}, 300);
+    const dataFrame: DataFrame = data?.slice(startIndex, endIndex);
+    return dataFrame;
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export const usePaginatedData = ({
+    current,
+    recordsPerPage,
+}: {
+    current: number;
+    recordsPerPage: number;
+}) => {
+    const [records, setRecords] = useState<DataFrame>();
+    const [pageSize, setPageSize] = useState(recordsPerPage);
+    const [page, setPage] = useState(current);
+
+    useEffect(() => {
+        (async () => {
+            const newRecords = await fetchData({
+                size: pageSize,
+                page,
+            });
+            setRecords(newRecords);
+        })();
+    }, [recordsPerPage, page, pageSize, setRecords]);
+
+    const paginatedData = { value: records, isLoading: false };
+    const totalRecords = data.values.length;
+
+    return {
+        records,
+        setRecords,
+        pageSize,
+        setPageSize,
+        page,
+        setPage,
+        totalRecords,
+        paginatedData,
+    };
+};
+
+export const PaginatedTemplate = (pagination: Pagination) => {
+    const { current = 1, recordsPerPage = 5 } = pagination;
+    const { paginatedData, page, pageSize, setPage } = usePaginatedData({
+        current,
+        recordsPerPage,
+    });
+
+    const stateFulOptions = {
+        ...options,
+        pagination: {
+            current: page,
+            recordsPerPage: pageSize,
+            totalRecords: data.length,
+            onPageChange: setPage,
+        },
+    };
+
+    return <Table data={paginatedData} options={stateFulOptions} />;
+};
+
+export const PageSizeTemplate = (pagination: Pagination) => {
+    const { current = 1, recordsPerPage = 5 } = pagination;
+    const { paginatedData, page, pageSize, setPage, setPageSize } = usePaginatedData({
+        current,
+        recordsPerPage,
+    });
+
+    const stateFulOptions = {
+        ...options,
+        pagination: {
+            current: page,
+            recordsPerPage: pageSize,
+            totalRecords: data.length,
+            onPageChange: setPage, //
+            pageSizeSelect: {
+                options: [
+                    { label: '2 / pages', value: 2 },
+                    { label: '5 / pages', value: 5 },
+                    { label: '10 / pages', value: 10 },
+                ],
+                onChange: (newSize: number) => {
+                    setPageSize(newSize);
+                    setPage(1);
+                }, // stateful, defined in template
+            },
+        },
+    };
+
+    return <Table data={paginatedData} options={stateFulOptions} />;
+};
+
+/* This template will fail to catch a new page and returns previous  data: {
+      value,
+      loading: false,
+  }, page and pageSize
+simulating e.g. an API call fail.
+The select should stay on it's previous value, not the clicked one.
+*/
```

Comment:
IMO, we should build an optimistic component that doesn't handle case of failure. 
It up to the parent that use the table to prevent invalid prop configuration.

Reply:
It's a bit different in the case of a native select: if the user selects a value, the value will be set in the HTML with no pass into the JS at all. So if the user clicks on '10' but say the API fail etc. all the Svelte app will remain the same (say on 5), but the select will still display 10.

Reply:
OK I get it, that explain: 
```
    $: if (value !== selected) {
        value = selected;
    }
```